### PR TITLE
Replace custom gist implementation with github rendering

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Gist replacer

### Description
This PR replaces our custom gist frame implementation with the github embedding url (like https://gist.github.com/ErikMichelson/1fcd4867c3740dc1c9e3e78eb6dc2070.pibb).

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: src/components/markdown-renderer/replace-components/gist/use-resize-gist-frame.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { useCallback, useEffect, useRef, useState } from 'react'
+
+const isLeftMouseButtonClicked = (mouseEvent: MouseEvent): boolean => {
```
Comment (ErikMichelson) [inline src/components/markdown-renderer/replace-components/gist/use-resize-gist-frame.ts:9]: Missing esdoc

File: src/components/markdown-renderer/replace-components/gist/use-resize-gist-frame.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  return mouseEvent.buttons === 1
+}
+
+/**
+ * Extracts the absolute horizontal position of the mouse or touch point from the event.
+ * If no position could be found or
```
Comment (ErikMichelson) [inline src/components/markdown-renderer/replace-components/gist/use-resize-gist-frame.ts:15]: Incomplete esdoc

**Comment by ErikMichelson** [comment]
In the CHANGELOG should be documented, that links from within the Gist frame must be opened in a new tab (Ctrl+Click, mouse wheel click, right-click -> open in new tab).

**Comment by mrdrogdrog** [comment]
done

**Review by ErikMichelson — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]

**Pull request merged**